mathics/builtin/calculus.py

```diff
@@ -604,11 +607,19 @@ def apply(self, f, xs, evaluation):
             # e.g. NotImplementedError: Result depends on the sign of
             # -sign(_Mathics_User_j)*sign(_Mathics_User_w)
             return
-
         if prec is not None and isinstance(result, sympy.Integral):
             # TODO MaxExtaPrecision -> maxn
             result = result.evalf(dps(prec))
-        result = from_sympy(result)
+        else:
+            result = from_sympy(result)
+        # If the result is defined as a Piecewise expression,
+        # use ConditionalExpression.
+        # This does not work now because the form sympy returns the values
+        if result.get_head_name() == "System`Piecewise":
```

Comment:
Another small thing. I think the idiom typically used is `result.is_form()` right?

Reply:
Just to have in mind, has_form is a method of `Expression`, so to use it, we need to check first that the object is not an atom...

Reply:
even checking with is_atom() the method fails for strings. For this reason, I reverted this to the previous strategy. Maybe later a better approach comes up.

Reply:
Here is a suggestion: add the `is_form()`  method to Atom. The implementation is pretty simple: return False.

Reply:
That is a great idea! I implemented it in `BaseExpression` instead of `Symbol`.